lib/colorado_lottery.rb

```diff
@@ -0,0 +1,52 @@
+class ColoradoLottery
+
+  attr_reader :registered_contestants,
+              :winners,
+              :current_contestants
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants= Hash.new { |hash, key| hash[key] = [] }
+  end
+
+  def interested_and_18?(contestant, game)
+    interested = contestant.game_interests.any? { |interested| interested == game.name }
+    interested && contestant.age >= 18
+  end
+
+  def can_register?(contestant, game)
+    in_state_or_national = (!contestant.out_of_state? || game.national_drawing)
+    interested_and_18?(contestant, game) && in_state_or_national
+  end
+
+  def register_contestant(contestant, game)
+    if can_register?(contestant, game)
+      @registered_contestants[game.name] = [] if !@registered_contestants[game.name]
```

Comment:
cool-- i prefer

`@registered_contestants[game.name] = [] if @registered_contestants[game.name].nil?`

it a) keeps things framed in the positive and b) explicitly checks for nil, rather than just `falsey`